tracing/src/test/java/com/palantir/tracing/AsyncSlf4jSpanObserverTest.java

```diff
@@ -219,6 +222,38 @@ public void testSpanMetadataConvertsToZipkinBinaryAnnotations() {
         assertThat(annotationMap).isEqualTo(spanMetadata);
     }
 
+    @Test
+    public void testCompleteSpanAfterExecutorShutdown() {
+        ExecutorService executor = MoreExecutors.newDirectExecutorService();
+        Tracer.subscribe("", AsyncSlf4jSpanObserver.of(
+                "serviceName", Inet4Address.getLoopbackAddress(), logger, executor));
+        OpenSpan span1 = Tracer.startSpan("operation");
+        assertThat(Tracer.completeSpan()).isPresent().get()
+                .extracting("spanId", "operation")
+                .contains(span1.getSpanId(), span1.getOperation());
+
+        OpenSpan span2 = Tracer.startSpan("operation");
+        executor.shutdown();
+        assertThat(Tracer.completeSpan()).isPresent().get()
+                .extracting("spanId", "operation")
```

Comment:
Missing a tracer.unsubscribe. If handling works as expected that shouldn’t be allowed to break anything, but may cause confusing logging depending on test order

Reply:
moved the `unsubscribe`s to `before`/`after` to ensure clean state